orchestrator/workflow.py

```diff
@@ -1157,11 +1363,18 @@ def _logstep(*x: Any) -> Process:
     logger.bind(workflow=pstat.workflow.name)
 
     def resume_suspend(process: Process) -> Process:
-        step = steps.pop(0)
+        state = process.unwrap()
+        if "__step_group" in state:
+            step = steps[0]
+        else:
+            step = steps.pop(0)
         return _logstep(step, process)
 
     next_state = pstat.state.resume(resume_suspend)
-
+    # Set the step_log_fn in the contextvar.
+    # This enables recursive step execution of sub steps with the same StepLogFunc.
+    # Should probably be refactored at some point as contextvars is a kind of global state.
+    step_log_fn_var.set(_logstep)
```

Comment:
It is global state but also completely isolated, so it's okay-ish.

I do think it would be wise to clear the contextvar after the stepgroup is done; I don't think we do that at the moment?